packages/editor/src/components/plugins-block-settings-menu-group/index.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createSlotFill } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
+
+const { Fill: PluginsBlockSettingsMenuGroup, Slot } = createSlotFill( 'PluginsBlockSettingsMenuGroup' );
+
+PluginsBlockSettingsMenuGroup.Slot = ( { fillProps } ) => {
```

Comment:
It feels a little odd to have `PluginsBlockSettingsMenuGroup` and `PluginBlockSettingsMenuItem` being parts of different packages - could you please clarify the rationale behind that decision?

Reply:
`PluginBlockSettingsMenuItem` is just one of the implementation of the `PluginsBlockSettingsMenuGroup` fill. In fact, it is a wrapper which enforces consistency across all menu items added in `edit-post`. My reasoning is that this slot is general purpose for all `editor` implementations. The same applies for other slots like `InspectorControls` and `BlockControls` - they only define the place where plugins can put they fills, they don't have any opinions. You could use them directly as it happens with blocks and their definitions, but using an abstraction will help us to apply those block settings only in `edit-post` context. It is still possible to use `PluginsBlockSettingsMenuGroup` fill directly, but this is going to be a bit more complex.

I'm happy to discuss it further, this is something that needs some more thought as this is something that might have implications in the future if we don't limit its usage by scoping it.

For the better context, we already did some exploration for `edit-template` page which would allow you to use blocks to build your own page template. This is why I'm not sold on the idea of having specialized implementations inside `editor` package as they might show up in the context some plugin developers didn't intend to. It's safer to assume all plugins are applied to `edit-post` as of today and move it up to `editor` in the future than the other way around.